homeassistant/components/device_automation/toggle_entity.py

```diff
@@ -0,0 +1,193 @@
+"""Device automation helpers for toggle entity."""
+import voluptuous as vol
+
+import homeassistant.components.automation.state as state
+from homeassistant.components.device_automation.const import (
+    CONF_IS_OFF,
+    CONF_IS_ON,
+    CONF_TOGGLE,
+    CONF_TURN_OFF,
+    CONF_TURN_ON,
+)
+from homeassistant.core import split_entity_id
+from homeassistant.const import (
+    CONF_CONDITION,
+    CONF_DEVICE,
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_ENTITY_ID,
+    CONF_PLATFORM,
+    CONF_TYPE,
+)
+from homeassistant.helpers.entity_registry import async_entries_for_device
+from homeassistant.helpers import condition, config_validation as cv, service
+
+ENTITY_ACTIONS = [
+    {
+        # Turn entity off
+        CONF_DEVICE: None,
+        CONF_TYPE: CONF_TURN_OFF,
+    },
+    {
+        # Turn entity on
+        CONF_DEVICE: None,
+        CONF_TYPE: CONF_TURN_ON,
+    },
+    {
+        # Toggle entity
+        CONF_DEVICE: None,
+        CONF_TYPE: CONF_TOGGLE,
+    },
+]
+
+ENTITY_CONDITIONS = [
+    {
+        # True when entity is turned off
+        CONF_CONDITION: "device",
+        CONF_TYPE: CONF_IS_OFF,
+    },
+    {
+        # True when entity is turned on
+        CONF_CONDITION: "device",
+        CONF_TYPE: CONF_IS_ON,
+    },
+]
+
+ENTITY_TRIGGERS = [
+    {
+        # Trigger when entity is turned off
+        CONF_PLATFORM: "device",
+        CONF_TYPE: CONF_TURN_OFF,
+    },
+    {
+        # Trigger when entity is turned on
+        CONF_PLATFORM: "device",
+        CONF_TYPE: CONF_TURN_ON,
+    },
+]
+
+ACTION_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_DEVICE): None,
+        vol.Optional(CONF_DEVICE_ID): str,
+        vol.Required(CONF_DOMAIN): str,
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Required(CONF_TYPE): vol.In([CONF_TOGGLE, CONF_TURN_OFF, CONF_TURN_ON]),
+    }
+)
+
+CONDITION_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_CONDITION): "device",
+        vol.Optional(CONF_DEVICE_ID): str,
+        vol.Required(CONF_DOMAIN): str,
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Required(CONF_TYPE): vol.In([CONF_IS_OFF, CONF_IS_ON]),
+    }
+)
+
+TRIGGER_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_PLATFORM): "device",
+        vol.Optional(CONF_DEVICE_ID): str,
+        vol.Required(CONF_DOMAIN): str,
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Required(CONF_TYPE): vol.In([CONF_TURN_OFF, CONF_TURN_ON]),
+    }
+)
+
+
+def _is_domain(entity, domain):
+    return split_entity_id(entity.entity_id)[0] == domain
+
+
+async def async_action_from_config(hass, config, variables, context, domain):
```

Comment:
Can we rename this method to be `async_call_action_from_config`, otherwise there is no verb in the function and it sounds weird.

Reply:
Fixed.